### ISSUE TITLE
make fails

### ISSUE THREAD

**Issue by krtschmr** [beatgammit/pam-http#2]
i cloned the git and run `make` which then fails

```
tim@vm:~/pam/pam-http$ make
cc -Werror -Wall -o test src/test.c -lpam -lpam_misc
src/test.c:12:31: fatal error: security/pam_appl.h: No such file or directory
compilation terminated.
Makefile:11: recipe for target 'test' failed
make: *** [test] Error 1

```

how can i install this for testing?

**Comment by DaEkstrim** [comment]
Most probably you solved it by now; but for future readers take a look at this link: http://ask.xmodulo.com/fatal-error-security-pam-modules.html

However this project would definitely benefit from better documentation on the kind of request the server receives (GET, POST, Query Params?, Body Params?, etc)

**Comment by krtschmr** [comment]
no, we didn't solved this. maybe we don't need this anymore as we trying a different solution now. instead of ftp-auth with user-accounts (same logins for our website) we just use cloud.

**Comment by beatgammit** [comment]
@krtschmr Sorry for the late reply, but late is better than never, right?

> However this project would definitely benefit from better documentation on the kind of request the server receives

Eh, this was mostly a proof of concept and wasn't intended to be used as-is. I had grand aspirations of doing something awesome like implementing a portable profile (login from anywhere and have all your things), but that ended up in the waste basket along with many other dreams of taking over the world ;)

> fatal error: security/pam_appl.h: No such file or directory

Do you have pam installed? It works fine for me on Arch Linux, so if you gave me some more information about your OS, I might be able to provide more assistance.

**Comment by abarrafo** [comment]
I saw this issue on Centos/Amazon Linux. simply run `yum install pam-devel` to install the necessary dependencies.
